### PR TITLE
[PKG-1486] Update to v11.0.0

### PR DESCRIPTION
**Pull request by schuylermartin45**
- Updates to version 11.0.0
- Adds some clarifying text around the relationship with `pyarrow-feedstock`
- Suppresses linter errors that are not relevant to a CPP binary. These are documented in the `meta.yaml`.

Checklist
* [ ] Used a fork of the feedstock to propose changes
* [ ] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/conda_smithy.html#how-to-re-render ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by schuylermartin45** [comment]
This change is directly tied to a change in [pyarrow](https://github.com/AnacondaRecipes/pyarrow-feedstock/pull/14)

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,6 +1,10 @@
-{% set version = "10.0.1" %}
+# Note that this feedstock must be paired with bumping the version for
+# `pyarrow-feedstock` as well. The two are tightly coupled. This is the CPP
+# component/backend, `pyarrow` has the Python libraries. Also, the SHA-256
+# hashes should match.
+{% set version = "11.0.0" %}
 {% set filename = "apache-arrow-" + version + ".tar.gz" %}
-{% set checksum = "c814e0670112a22c1a6ec03ab420a52ae236a9a42e9e438c3cbd37f37e658fb3" %}
+{% set sha256 = "2dd8f0ea0848a58785628ee3a57675548d509e17213a2f5d72b0d900b43f5430" %}
```
Comment (schuylermartin45) [inline recipe/meta.yaml:7]: `sha256` seems to be the more common name from what I've seen, so I changed it to be "more consistent"

**Review by ELundby45 — APPROVED** [review]
LGTM

**Review by M-Waszkiewicz-Anaconda — APPROVED** [review]
LGTM

**Pull request merged**